pkg/controllers/rollout/controller.go

```diff
@@ -732,96 +716,105 @@ func (r *Reconciler) SetupWithManagerForClusterResourceBinding(mgr runtime.Manag
-			CreateFunc: func(ctx context.Context, e event.CreateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
-				klog.V(2).InfoS("Handling a resourceBinding create event", "resourceBinding", klog.KObj(e.Object))
-				enqueueResourceBinding(e.Object, q)
-			},
-			UpdateFunc: func(ctx context.Context, e event.UpdateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
-				klog.V(2).InfoS("Handling a resourceBinding update event", "resourceBinding", klog.KObj(e.ObjectNew))
-				handleResourceBindingUpdated(e.ObjectNew, e.ObjectOld, q)
-			},
-			GenericFunc: func(ctx context.Context, e event.GenericEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
-				klog.V(2).InfoS("Handling a resourceBinding generic event", "resourceBinding", klog.KObj(e.Object))
-				enqueueResourceBinding(e.Object, q)
-			},
-		}).
+		Watches(&placementv1beta1.ResourceOverride{}, resourceOverrideHandlerFuncs(false)).
+		Watches(&placementv1beta1.ResourceBinding{}, bindingHandlerFuncs()).
 		// Aside from resource snapshot and binding objects, the rollout
 		// controller also watches ResourcePlacement objects,
 		// so that it can push apply strategy updates to all bindings right away.
-		Watches(&placementv1beta1.ResourcePlacement{}, handler.Funcs{
-			// Ignore all Create, Delete, and Generic events; these do not concern the rollout controller.
-			UpdateFunc: func(ctx context.Context, e event.UpdateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
-				handlePlacement(e.ObjectNew, e.ObjectOld, q)
-			},
-		}).
+		Watches(&placementv1beta1.ResourcePlacement{}, placementHandlerFuncs()).
 		Complete(r)
 }
 
+// resourceSnapshotObjHandlerFuncs returns the handler functions for resource snapshot obj events
+// (including cluster resource snapshots and resource snapshots).
+func resourceSnapshotObjHandlerFuncs() handler.Funcs {
+	return handler.Funcs{
+		CreateFunc: func(ctx context.Context, e event.CreateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+			klog.V(2).InfoS("Handling a cluster resource snapshot create event", "resourceSnapshot", klog.KObj(e.Object))
+			handleResourceSnapshot(e.Object, q)
+		},
+		GenericFunc: func(ctx context.Context, e event.GenericEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+			klog.V(2).InfoS("Handling a cluster resource snapshot generic event", "resourceSnapshot", klog.KObj(e.Object))
+			handleResourceSnapshot(e.Object, q)
+		},
+	}
+}
+
+// resourceOverrideSnapshotHandlerFuncs returns the handler functions for resource override snapshot events.
+func resourceOverrideSnapshotHandlerFuncs(enqueueCRP bool) handler.Funcs {
+	return handler.Funcs{
+		CreateFunc: func(ctx context.Context, e event.CreateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+			klog.V(2).InfoS("Handling a resourceOverrideSnapshot create event", "resourceOverrideSnapshot", klog.KObj(e.Object))
```

Comment:
add the enqueueCRP bool to the log? so that we can distinguish whether the log is for CRP or RP